src/gulpfile.js

```diff
@@ -186,74 +186,56 @@ gulp.task('chat-styles', ['chat-styles-handlebars', 'chat-styles-js'], function(
 
 // Compile and concatenate handlebars files for all chat styles.
 gulp.task('chat-styles-handlebars', function() {
-    var promises = [];
-    fs.readdir(config.chatStylesPath, function(err, list){
-        if(err) return done(err);
-        list.filter(function(path){
-            return fs.lstatSync(config.chatStylesPath + "/" + path).isDirectory();
-        }).map(function(dir){
-            promises.push(new Promise(function(resolve, reject){
-                var pipeline = gulp.src(config.chatStylesPath + '/' + dir + '/templates_src/client_side/**/*.handlebars')
-                  .pipe(handlebars({
-                      // Use specific version of Handlebars.js
-                      handlebars: handlebarsEngine
-                  }))
-                  .pipe(wrapHandlebarsTemplate())
-                  .pipe(concat('templates.js'))
-                  .pipe(uglify({preserveComments: 'some'}))
-                  .pipe(header(config.compiledTemplatesHeader))
-                  .pipe(gulp.dest(config.chatStylesPath + '/' + dir + '/templates_compiled/client_side'));
-                pipeline.on('end', resolve);
-                pipeline.on('error', reject);
-            }));
+    return Promise.all(getChildDirs(config.chatStylesPath).map(function (dir) {
```

Comment:
I agree about `getChildDirs` must return a promise, but we cannot rid of `Array.prototype.map` because we still have to convert an array of dirs (no matter how it was got) into gulp pipes.